Sources/NIO/EventLoopFuture.swift

```diff
@@ -1325,3 +1325,78 @@ extension EventLoopFuture {
         return self
     }
 }
+
+// MARK: unwrap
+
+extension EventLoopFuture {
+    /// Unwrap an `EventLoopFuture` where its type parameter is an `Optional`.
+    ///
+    /// Unwrap a future returning a new `EventLoopFuture`. When the resolved future's value is `Optional.some(...)`
+    /// the new future is created with the identical value. Otherwise the `Error` passed in the `or` parameter
+    /// is thrown. For example:
+    /// ```
+    /// do {
+    ///     try promise.futureResult.unwrap(or: ErrorToThrow).wait()
+    /// } catch ErrorToThrow {
+    ///     ...
+    /// }
+    /// ```
+    ///
+    /// - parameters:
+    ///     - or: the `Error` that is thrown when then resolved future's value is `Optional.none`.
+    /// - returns: an new `EventLoopFuture` with new type parameter `T` and the same value as the resolved
+    ///     future.
+    /// - throws: the `Error` passed in the `or` parameter when the resolved future's value is `Optional.none`.
+    func unwrap<NewValue>(or error: Error) -> EventLoopFuture<NewValue> where Value == Optional<NewValue> {
```

Comment:
Can we add an `@inlinable` declaration to these three methods, please?

Reply:
I'm a little uncomfortable with these overloads; I think `unwrap(orError:)` and `unwrap(orElse:)` would be more readable and avoid any ambiguity in the case of `Value` also being an error.

Reply:
(Maybe `orElse` isn't necessary if we have `orError`)

Reply:
Oh good catch, that could _absolutely_ be awkward.